pkg/monitoring/migration/prometheus.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package migration
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	k8sv1 "k8s.io/api/core/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+)
+
+type gaugeAction bool
+
+const increase gaugeAction = true
+const decrease gaugeAction = false
+
+var (
+	migrationsLabels = []string{"vmi", "source", "target"}
+
+	CurrentPendingMigrations = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "kubevirt_migrations_pending",
+			Help: "Number of current pending migrations.",
+		},
+		migrationsLabels,
+	)
+
+	CurrentSchedulingMigrations = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "kubevirt_migrations_scheduling",
+			Help: "Number of current scheduling migrations.",
+		},
+		migrationsLabels,
+	)
+
+	CurrentRunningMigrations = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "kubevirt_migrations_running",
+			Help: "Number of current running migrations.",
+		},
+		migrationsLabels,
+	)
+
+	MigrationsSucceededTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "kubevirt_migrations_succeeded_total",
+			Help: "Number of migrations successfully executed.",
+		},
+		migrationsLabels,
+	)
+
+	MigrationsFailedTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "kubevirt_migrations_failed_total",
```

Comment:
@machadovilaca Base on the comment above I would update the metric name to kubevirt_migrate_vmi_failed_total or kubevirt_vmi_migrate_failed_total

Reply:
updated